rmgpy/rmg/main.py

```diff
@@ -1463,21 +1481,126 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
 
     def initializeReactionThresholdAndReactFlags(self):
         numCoreSpecies = len(self.reactionModel.core.species)
-        if self.filterReactions:
-            self.unimolecularReact = np.zeros((numCoreSpecies),bool)
-            self.bimolecularReact = np.zeros((numCoreSpecies, numCoreSpecies),bool)
-            self.unimolecularThreshold = np.zeros((numCoreSpecies),bool)
+
+        # Initialize everything to react by default, but we will handle the restart and filtering case immediately after
+        self.unimolecularReact = np.ones(numCoreSpecies, bool)
+        self.bimolecularReact = np.ones((numCoreSpecies, numCoreSpecies), bool)
+        if self.trimolecular:
+            self.trimolecularReact = np.ones((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
+
+        if self.filterReactions or self.restart:  # Otherwise no need to initialize thresholds or fix react flags
+            self.unimolecularThreshold = np.zeros(numCoreSpecies,bool)
             self.bimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies),bool)
             if self.trimolecular:
-                self.trimolecularReact = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
-                self.trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
-        else:
-            # By default, react everything
-            self.unimolecularReact = np.ones((numCoreSpecies),bool)
-            self.bimolecularReact = np.ones((numCoreSpecies, numCoreSpecies),bool)
-            if self.trimolecular:
-                self.trimolecularReact = np.ones((numCoreSpecies, numCoreSpecies, numCoreSpecies),bool)
-            # No need to initialize reaction threshold arrays in this case
+                self.trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies),bool)
```

Comment:
Spaces after commas.

Reply:
Thanks for the catch! I forgot that I could change this since I was changing this line anyways